src/store/staking/staking.sagas.ts

```diff
@@ -128,13 +131,61 @@ export function* fetchStakesList() {
   }
 }
 
+export function* fetchFilteredHistoryStakeEvents(contractAddresses: string[]) {
+  const subgraphClient = yield* select(subgraphClientSelector);
+
+  if (!subgraphClient) throw new Error('Wallet not connected');
+
+  const { stakeEvents } = yield* call(historyStakesQuery, subgraphClient, {
```

Comment:
For me it could be done in the main saga. I'm not sure if saga with only one call is worth extracting